odxtools/basicstructure.py

```diff
@@ -107,10 +107,15 @@ def print_free_parameters_info(self) -> None:
         print(parameter_info(self.free_parameters), end="")
 
     def convert_physical_to_internal(self,
-                                     param_values: ParameterValueDict,
+                                     param_values: ParameterValue,
```

Comment:
param_values or param_value ?

Reply:
both, but this is a bit twisted so let me explain: `ParameterValue` represents the value which can be expressed by any parameter while `ParameterValueDict` is a name->value dictionary for a set of parameters. That said, there are composite parameters (structures) which expect a name-to-value dictionary as their value, so a `ParameterValue` object is nested and can thus be a `ParameterValueDict`, which is exactly what is expected here.

unfortunately (or maybe luckily), we cannot restrict the signature of methods of child classes relative to their parents, so we need to use `ParameterValue` as the type annotation and ensure at runtime that it is the right kind of object, i.e., a dictionary...

Reply:
okay, I switched it to singular.